server/src/main/java/io/pravega/schemaregistry/storage/impl/groups/PravegaKeyValueGroups.java

```diff
@@ -108,7 +108,7 @@ public PravegaKeyValueGroups(TableStore tableStore, ScheduledExecutorService exe
         String namespace = nameSpace == null ? "" : nameSpace;
         ByteBuf continuationToken;
         if (token == null || token.equals(ContinuationToken.EMPTY)) {
-            continuationToken = IteratorState.EMPTY.toBytes();
+            continuationToken = IteratorStateImpl.EMPTY.getToken();
```

Comment:
This appears to be leaking the internal details of Pravega client here. Is it avoidable? For example, if there was a factory that returned an object of the interface, depending directly in the `IteratorStateImpl` could be avoided.

Reply:
there are two points here -- 
1. identity for continuation token would then have to be created from empty ByteBuffer and pass it around. which is equivalent. this at least keeps that opaque.
2. this is an an internal hidden away object. the IteratorStateImpl is a public object and Empty can be used to express the state identity.

i can use empty bytebuffer as the identity too if that is preferred as that relies on the interface rather than the impl.

Reply:
Yeah, I'd say doing away with this dependency on an internal implementation is better, especially as it is not necessary for any other functionality here.

Reply:
The controller's SegmentHelper class actually requires IteratorStateImpl instead of IteratorState interface now.

https://github.com/pravega/pravega/blob/37c5082bfe248b2682826f4658509aa1b3b4aaf7/controller/src/main/java/io/pravega/controller/server/SegmentHelper.java#L489
3:28

So we need to create and pass an object of Impl.